code/modules/GOAP/cyka_blyat/actions.dm

```diff
@@ -0,0 +1,339 @@
+	return TRUE
+
+/datum/goap_action/russian/medic/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	if(target)
+		A.say(pick(list("Isprav'te sebya!","Medik na stsene!","Ostanovite poluchat' vystrel tak mnogo!")))
+		if(prob(1))
+			A.visible_message("[A] hands [target] some vodka! [target] chugs it, feeling rejuvenated!")
+		else
+			A.visible_message(pick(list("[A] applies some libital to [target]!", "[A] bandages some wounds on [target]!", "[A] quickly stitches close a cut on [target]!")))
+		var/mob/living/simple_animal/hostile/russian/ranged/R = target
+		R.health += (R.maxHealth/4)
+		action_done = TRUE
+		return ..()
+	else
+		action_done = TRUE
+		return FALSE
+
+/datum/goap_action/russian/medic/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/resupply
+	name = "Resupply Allies"
+	cost = 1
+	cooldown = 10
+
+/datum/goap_action/russian/resupply/New()
+	..()
+	preconditions = list()
+	preconditions["allyRearmed"] = FALSE
+	effects = list()
+	effects["allyRearmed"] = TRUE
+
+/datum/goap_action/russian/resupply/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/simple_animal/hostile/russian/ranged/R in viewl)
+		if(!R.stat)
+			if(!R.reloads_left)
+				target = R
+				break
+	return (target != null)
+
+/datum/goap_action/russian/resupply/RequiresInRange()
+	return FALSE
+
+/datum/goap_action/russian/resupply/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	var/mob/living/simple_animal/hostile/russian/ranged/RU = target
+	if(target)
+		A.say(pick(list("Voz'mi eto boyepripasy, tovarishch!","Perezagruzite oruzhiye!","Perezagruzite svoy grebanyy idiot!")))
+		A.visible_message("[A] throws a new magazine to [target]!")
+		var/obj/item/russian_reload/RE = new /obj/item/russian_reload(get_turf(AA_MATCH_TARGET_OVERLAYS))
+		RE.throw_at(target, 10, 1)
+		RU.reloads_left = 3
+		RU.say(pick(list("Spasiba!","Spasiba za boyepripasy!","Ti khahroshiy spetsiahlist!")))
+		QDEL_IN(RE, 15)
+		action_done = TRUE
+		return ..()
+	else
+		action_done = TRUE
+		return FALSE
+
+
+/datum/goap_action/russian/resupply/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/dodge
+	name = "Dodge!"
+	cost = 1
+	cooldown = 50
+	saytext = "OPAH! CANNOT EVEN HIT RUSSIAN BABY!"
+
+/datum/goap_action/russian/dodge/New()
+	..()
+	preconditions = list()
+	preconditions["dodgeEnemy"] = FALSE
+	effects = list()
+	effects["dodgeEnemy"] = TRUE
+
+/datum/goap_action/russian/dodge/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/carbon/C in viewl)
+		if(C && !C.stat)
+			target = C
+	return (target != null)
+
+/datum/goap_action/russian/dodge/RequiresInRange(atom/agent)
+	return FALSE
+
+/datum/goap_action/russian/dodge/Perform(mob/living/simple_animal/hostile/H)
+	H.SpinAnimation(5, 0) // half a second spin but one second immunity
+	DodgeEffect(H)
+	var/oldhealth = H.health
+	H.health = INFINITY // Iframes!
```

Comment:
note that I'm pretty sure that some instant-kill effects will still kill things with an infinite amount of health (check with a wand of death?)

Reply:
If they're getting hit by a wand of death I think just let them die lmao

Reply:
But isn't the fluff of this ability that they're supposed to be dodging the projectile, not getting hit by it and ignoring its effects?

They'd probably also get knocked back by meteorshot rounds as well.

Would using the godmode flag instead of temporarily giving them infinite health work better?

Reply:
They’d still be hit though but a godmode flag might work better